src/main.py

```diff
@@ -0,0 +1,122 @@
+
+    logging.info('Create the train and validation datasets.')
+    train_dataset = _get_instance(src.data.datasets, config.dataset, 'train')
+    valid_dataset = _get_instance(src.data.datasets, config.dataset, 'valid')
+
+    logging.info('Create the train and validation dataloaders.')
+    cls = getattr(src.data.datasets, config.dataset.name)
+    config.dataloader.kwargs.update(collate_fn=getattr(cls, 'collate_fn', None))
+    train_dataloader = _get_instance(src.data.dataloader, config.dataloader, train_dataset)
+    valid_dataloader = _get_instance(src.data.dataloader, config.dataloader, valid_dataset)
+
+    logging.info('Create the network architecture.')
+    net = _get_instance(src.model.nets, config.net)
+
+    logging.info('Create the loss functions and the corresponding weights.')
+    losses, loss_weights = [], []
+    defaulted_losses = [loss for loss in dir(torch.nn) if 'Loss' in loss]
+    for config_loss in config.losses:
+        if config_loss.name in defaulted_losses:
+            loss = _get_instance(torch.nn, config_loss)
+        else:
+            loss = _get_instance(src.model.losses, config_loss)
+        losses.append(loss)
+        loss_weights.append(config_loss.weight)
+
+    logging.info('Create the metric function.')
+    metrics = [_get_instance(src.model.metrics, config_metric) for config_metric in config.metrics]
+
+    logging.info('Create the optimizer.')
+    optimizer = _get_instance(torch.optim, config.optimizer, net.parameters())
+
+    logging.info('Create the learning rate scheduler.')
+    lr_scheduler = _get_instance(torch.optim.lr_scheduler, config.lr_scheduler, optimizer) if config.get('lr_scheduler') else None
+
+    logging.info('Create the logger.')
+    config.logger.kwargs.update(log_dir=config.main.saved_dir + '/log', net=net)
+    logger = _get_instance(src.callbacks.loggers, config.logger)
+
+    logging.info('Create the monitor.')
+    config.monitor.kwargs.update(root=config.main.saved_dir + '/checkpoints')
+    monitor = _get_instance(src.callbacks.monitor, config.monitor)
+
+    logging.info('Create the trainer.')
+    kwargs = {'device': device,
+              'train_dataloader': train_dataloader,
+              'valid_dataloader': valid_dataloader,
+              'net': net,
+              'losses': losses,
+              'loss_weights': loss_weights,
+              'metrics': metrics,
+              'optimizer': optimizer,
+              'lr_scheduler': lr_scheduler,
+              'logger': logger,
+              'monitor': monitor}
+    config.trainer.kwargs.update(kwargs)
+    trainer = _get_instance(src.runner.trainers, config.trainer)
+
+    # Start training.
+    logging.info('Start training.')
+    trainer.train()
+    logging.info('End training.')
+
+
+def _parse_args():
+    parser = argparse.ArgumentParser(description="The script to train.")
+    parser.add_argument('--config_file', '-c', type=Path, help='The path of the config file.')
```

Comment:
Need to set required = True?

Reply:
Yes, or `parser.add_argument('config_file', type=Path, help='The path of the config file.')`?

Reply:
Both can work.

Reply:
Prefer the last one, have updated.